group1/communication/node/main.js

```diff
@@ -29,27 +29,46 @@ var done = false;
 app.use(multer({
         'dest': '../../comm/static/uploads',
         onFileUploadStart: function(file) {
+            if(testNameValidation(file.originalname)){
                 console.log(file.originalname + ' is starting...');
+            } else {
+                console.log('File name is invalid');
+                return false;
+            }
         },
         onFileUploadComplete: function(file) {
                 console.log(file.fieldname + ' uploaded to ' + file.path);
                 done = true;
         }
 }));
 
+function testNameValidation(text) {
+    var format = /[ !@#$%^&*()+\-=\[\]{};':"\\|,<>\/?]/;
+    if(text.trim() == null || text.trim() == "" || format.test(text) == true) {
+        return false;
+    } else {
+        return true;
+    }
+}
+
 app.post('/upload', function(req,res){
         if (done==true) {
                 var hash = crypto.randomBytes(20).toString('hex');
-				var directory =  path.join(homedir + '/sites/www.3blueprints.com/static/uploads/');
+                console.log(path);
+				var directory =  path.join(homedir + '/BU_MET_CS673/group1/comm/static/uploads/');
+                console.log(directory);
```

Comment:
Does this logging have a purpose outside of debugging?  If so, can more information be added to provide context and clarity?  Otherwise, maybe it can be removed?